Add few shell filters

fabiobaltieri opened the pull request:
Add few shell filter functions, i2c, regulator and dac.

fabiobaltieri committed:
i2c_shell: add device filtering

Add shell device name filtering using DEVICE_API_IS.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
regulator_shell: add device filtering

Add shell device name filtering using DEVICE_API_IS.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
dac_shell: add device filtering

Add shell device name filtering using DEVICE_API_IS.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

yishai1999 commented on drivers/dac/dac_shell.c:
```diff
@@ -94,14 +94,31 @@ static int cmd_write_value(const struct shell *sh, size_t argc, char **argv)
 
+static bool device_is_dac(const struct device *dev)
+{
+	return device_is_ready(dev) && DEVICE_API_IS(dac, dev);
```
shell_device_filter already checks device_is_ready so this is superfluous

fabiobaltieri replied:
Hey you are right! I had copy pasted this from other drivers, removed from all of these and added an extra patch to clean up the existing ones. Thanks!

fabiobaltieri committed:
drivers: drop few redundant device_is_ready

Drop few redundant device_is_ready for functions that are only used as
argument to shell_device_filter, as shell_device_filter checks for that
alrady.

Suggested-by: Yishai Jaffe <yishai1999@gmail.com>
Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

yishai1999 commented on drivers/stepper/stepper_shell.c:
```diff
@@ -69,7 +69,7 @@ static void print_callback(const struct device *dev, const enum stepper_event ev
 
 static bool stepper_device_check(const struct device *dev)
```
Maybe change the function name to `device_is_stepper` to match the rest?

fabiobaltieri replied:
yeah good idea, done

fabiobaltieri commented on drivers/i2c/i2c_shell.c:
```diff
@@ -336,9 +336,14 @@ static int cmd_i2c_speed(const struct shell *shell_ctx, size_t argc, char **argv
 
+static bool device_is_i2c(const struct device *dev)
+{
+	return DEVICE_API_IS(i2c, dev);
```
cool, done
